homeassistant/components/permobil/sensor.py

```diff
@@ -0,0 +1,224 @@
+"""Platform for sensor integration."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from datetime import timedelta
+import logging
+from typing import Any
+
+from mypermobil import (
+    BATTERY_AMPERE_HOURS_LEFT,
+    BATTERY_CHARGE_TIME_LEFT,
+    BATTERY_DISTANCE_LEFT,
+    BATTERY_INDOOR_DRIVE_TIME,
+    BATTERY_MAX_AMPERE_HOURS,
+    BATTERY_MAX_DISTANCE_LEFT,
+    BATTERY_STATE_OF_CHARGE,
+    BATTERY_STATE_OF_HEALTH,
+    ENDPOINT_LOOKUP,
+    RECORDS_SEATING,
+    USAGE_ADJUSTMENTS,
+    USAGE_DISTANCE,
+)
+
+from homeassistant import config_entries
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import (
+    PERCENTAGE,
+    UnitOfEnergy,
+    UnitOfLength,
+    UnitOfTime,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import BATTERY_ASSUMED_VOLTAGE, COORDINATOR, DOMAIN
+from .coordinator import MyPermobilCoordinator
+
+
+def setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the sensor platform."""
+
+
+_LOGGER = logging.getLogger(__name__)
+SCAN_INTERVAL = timedelta(seconds=50)
+
+
+@dataclass
+class PermobilRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    value_fn: Callable[[Any], Any]
+    item: str
+
+
+@dataclass
+class PermobilSensorEntityDescription(
+    SensorEntityDescription, PermobilRequiredKeysMixin
+):
+    """Describes Permobil sensor entity."""
+
+
+SENSOR_DESCRIPTIONS: tuple[PermobilSensorEntityDescription, ...] = (
+    PermobilSensorEntityDescription(
+        value_fn=lambda x: x,
+        item=BATTERY_STATE_OF_CHARGE,
+        key="state_of_charge",
+        name="Permobil Battery Charge",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.BATTERY,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    PermobilSensorEntityDescription(
+        value_fn=lambda x: x,
+        item=BATTERY_STATE_OF_HEALTH,
+        key="state_of_health",
+        name="Permobil Battery Health",
```

Comment:
Similar case, drop the "Permobil" from the entity name, however, add a `translation_key` instead and move the rest of the name into `strings.json` so the name is translated into a users language.